Micronaut 5 compability

lissam1 opened the pull request.

micronaut-build committed:
Update common files

lissam1 committed:
update to kotlin 2.2.20

lissam1 committed:
update micronaut-build to 8.0.0-M5

lissam1 committed:
update to jvm 21 baseline

burtbeckwith committed:
checkstyle

burtbeckwith committed:
convert java to kotlin in kotlin examples

tests were failing with error 'There are test sources present and no filters are applied, but the test task did not discover any tests to execute'

burtbeckwith committed:
JDK 21 minimum

burtbeckwith committed:
build fixes, cleanup

burtbeckwith committed:
test cleanup, fixes

burtbeckwith commented on buildSrc/src/main/groovy/io.micronaut.build.internal.oraclecloud-example-httpfunction.gradle:
```diff
@@ -0,0 +1,19 @@
+    implementation(projects.micronautOraclecloudFunctionHttp)
+    runtimeOnly(libs.fn.runtime)
+    runtimeOnly(mnLogging.slf4j.simple)
+    testImplementation(libs.oci.common.httpclient.jersey3)
```
This is one change I wasn't sure about - the HTTP function tests were failing because of a missing HTTP client. I tried it with the Micronaut client but that didn't help, so I used jersey3 - is this correct?

burtbeckwith commented on buildSrc/src/main/groovy/io.micronaut.build.internal.oraclecloud-example-function.gradle:
```diff
@@ -0,0 +1,17 @@
+    runtimeOnly(libs.fn.runtime)
+    runtimeOnly(mnLogging.slf4j.simple)
+    testImplementation(libs.fn.testing.junit4)
+    testImplementation(libs.oci.common.httpclient.jersey3)
```
This is one change I wasn't sure about - the function tests were failing because of a missing HTTP client. I tried it with the Micronaut client but that didn't help, so I used jersey3 - is this correct?

burtbeckwith commented on gradle/libs.versions.toml:
```diff
@@ -44,59 +41,51 @@ gson = "2.13.2"
 micronaut-gradle-plugin = { module = "io.micronaut.gradle:micronaut-gradle-plugin", version.ref="micronaut-gradle-plugin" }
 protobuf-gradle-plugin = { module = "com.google.protobuf:protobuf-gradle-plugin", version.ref = "protobuf" }
-
-#dummy dependencies for renovate
```
I deleted these  because the checklist includes removing versions for Groovy and Spock - will this be a problem for renovate?

burtbeckwith committed:
minor build cleanup

burtbeckwith committed:
use io.micronaut.minimal.application for examples

using io.micronaut.application brings in the NI plugin, which we only want to use for example-java

burtbeckwith committed:
forgot to delete these

burtbeckwith committed:
add compileOnly(mn.reactor) for JDK 25

burtbeckwith committed:
compileOnly -> implementation, was getting the same error in test compile

burtbeckwith committed:
address Sonar issues

burtbeckwith committed:
checkstyle

burtbeckwith committed:
make flaky test more reliable

burtbeckwith committed:
use Netty HTTP client instead of jersey3

burtbeckwith committed:
use latest build plugin

burtbeckwith committed:
use latest MN versions

burtbeckwith committed:
use try-with-resources in tests

burtbeckwith committed:
update build plugin

burtbeckwith committed:
use java-base and kotlin-base plugins

burtbeckwith committed:
fix test

burtbeckwith committed:
bump more versions

burtbeckwith committed:
test cleanup
